### ISSUE TITLE
Homey.ManagerBLE caching keeps connected state.

### ISSUE THREAD

**Issue by nklerk** [athombv/apps-sdk-issues#86]
After finding a bluetooth peripheral the state is cached as connected. even after a disconnect.
this results in devices that can't reconnect after a disconnect.

doesn't mather if the disconnect is caused by code (.disconnect()) of the device is turned off/on.
the only solution in such a case is restarting the app.

current workaround i'm using is bypassing the cache:

```
      const BLEdevice = await Homey.ManagerBLE.find(this.getData().id);

      //Workaround for BUGs in ManagerBLE, (Cashing Bug, Connection Bug)
      if (BLEdevice.__peripheral) {
        BLEdevice.__peripheral = null;
      }

      this.connectedBLEdevice = await BLEdevice.connect();
```

**Comment by Maxmudjon** [comment]
some problem

**Comment by mrlukasbos** [comment]
This issue has been resolved in Homey v6.0.0 which is currently available as an experimental release.
